### PR TITLE
Using java.time.Duration and java.utils.Optional

### PR DESCRIPTION
**Pull request by qux42**
Pull-Request for implementing #37. Using java.time.Duration instead of long, and using java.utils.Optional for optional values in java

### CODE REVIEW

**Comment by carueda** [comment]
Thanks for your work on this, I appreciate it.

Can you summarize the changes in this PR?  It seems it's not only addressing #37 but also generating `Optional` for the java output, and apparently there's no, say, a `--genOptional` flag to make this opt in (otherwise it would be a breaking change). Also, could you split this into separate PRs, one for the `Duration` aspect, and one other for the `Optional` one?

**Comment by qux42** [comment]
thanks for feedback, will do that :)

**Comment by qux42** [comment]
finally had a bit time to split the first feature out with added feature flags  (#44 ). 
I hope I find time to do the same with the Durations in short term.
With this I closing this pull-request
